### DISCUSSION TITLE
ESP-01S with Relay Board and Push Button Toggle

### DISCUSSION THREAD

**Discussion by ldellus** [RavenSystem/esp-homekit-devices#2126] [MEPLHAA Scripts]
Hello,

I have an ESP-01S connected to a relay board, which works fine from HomeKit, but I'd like to add a push button to toggle the relay manually. My MEPLHAA script is:
{
	"c":{
		"io": [[[0], 2], [[2], 6]]     #GPIO 0 is the relay, GPIO 2 is the push button. I soldered it between the ESP-01S GPIO 2 and ground
	},
	"a":[{
		"0": {"r":[[0]]},
		"1": {"r":[[0, 1]]},
		"b": [[2], [2, 0]]            # This is part I can't figure out. After power on, pressing the push button turns on the relay only once, then nothing.
	}]
}
Appreciate any guidance you can give.
Thanks,
Laurent

**Comment by Tellicious** [comment]
If it's a pushbutton, change `"b": [[2], [2, 0]]` to `"b": [[2]]`. Then if its just soldered between GPIO2 and GND without any pull-up, you should declare `"io": [[[0], 2], [[2], 6, 1 , 1]]` to turn on the internal pull-up and invert the button behavior (active when low)

**Reply by ldellus** [reply]
It works! thank you so much for the help. I thought about the pull-up but didn't realize I needed to declare the io.
Sharing the full script for the benefit of the community. A push button is soldered between GPIO 2 and GND.
{
  "c": { 
        "io": [[[0], 2], [[2], 6, 1 , 1]],
        "n": "esp-01s-switch"
  },
  "a": [{
        "0": { "r": [ [ 0, 0 ] ] },
        "1": { "r": [ [ 0, 1 ] ] },
        "b": [ [ 2 ] ]
  }]
}
